src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/Miscellaneous/AutoLayoutStructureTests.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class AutoLayoutStructureTests
+    {
+        [StructLayout(LayoutKind.Auto)]
+        public struct SomeTestStruct_Auto
+        {
+            public int i;
+        }
+
+        [STAThread]
```

Comment:
This isn't going to have any impact.  Does this test require running on an STA thread?

Reply:
Nope.

@jiangzeng All `STAThread` attributes should be removed from Marshal API tests. There will be some cases in the CoreCLR repo, but in the CoreFX tests, this should never occur.

Reply:
Fixed, see in next commit